### PR TITLE
Add CI checks for Go code.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

- Adds a GHA workflow which will run `go test`, `go fmt`, and `go vet` for each Go module located under `src/go`, auto-detecting the paths to the modules and the required version of Go.

##### Test Plan

CI passes on this PR.

### CODE REVIEW

**Comment by ilyam8** [comment]
Looks needlessly complex to me: I suggest implementing it like it was in [go.d.plugin repo](https://github.com/netdata/go.d.plugin/blob/master/.github/workflows/test.yml):

- One runner for all checks, adding a separate runner for each "check" is not needed. No pros, only cons. What was the reasoning for separate runners?
- I see no need in parsing go.mod files. You can add it when/if needed. Please do it the same way it was in the go.d.plugin repo.
- Also, I need the same steps I have in the go.d.plugin repo: build, vet, test formatting, do tests - no need to improvise.

**Review by ilyam8 — CHANGES_REQUESTED** [review]
please address https://github.com/netdata/netdata/pull/17066#issuecomment-1966156730. Needlessly complicated as it is now.

**Comment by ilyam8** [comment]
Failing tests are expected and don't indicate a problem. The issue is #16997, not everything was copied. Missing testdata will be added in #17064.

**Comment by Ferroin** [comment]
Given the discussions in the sync today, there is little value for local testing in having a global build target to build all the go components, and the possibility of having testing that the CMake tooling around our Go components works correctly  as part of the regular CI is highly contentious (I still contend that this is something we should be testing, and that the overhead will not be problematic), I will plan to add that in a separate PR at this point in the interest of not delaying this one any further.

File: .github/workflows/go-tests.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    outputs:
+      matrix: ${{ steps.get-version.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get install -y python3-packaging
+      - name: Get Go version and modules
+        id: get-version
+        run: .github/scripts/get-go-version.py
```
Comment (ilyam8) [inline .github/workflows/go-tests.yml:68]: We never skip this job, why?

Comment (Ferroin) [reply .github/workflows/go-tests.yml:68]: Put differently, this is the same reason that we always generate the build matrix for the build, docker, and packaging CI workflows.

**Pull request merged**